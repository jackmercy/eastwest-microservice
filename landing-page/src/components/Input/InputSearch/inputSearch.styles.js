import { createUseStyles } from 'react-jss';

import { rem } from '../../../utils/jss';
import COLORS from '../../../constants/COLORS';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';

export default createUseStyles({
  inputWrapper: {
    position: 'relative',
    '& > [class^=icon-]': {
      top: '50%',
      right: rem(16),
      fontSize: rem(16),
      cursor: 'pointer',
      position: 'absolute',
      transform: 'translateY(-50%)',
      color: COLORS.primary,
    },
  },
  input: { paddingRight: rem(40) },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    inputWrapper: {
      '& > [class^=icon-]': { right: rem(12) },
    },
    input: { paddingRight: rem(32) },
  },
});
