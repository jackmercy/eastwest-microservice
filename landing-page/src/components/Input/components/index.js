export { default as FieldLabel } from './FieldLabel';
export { default as FieldError } from './FieldError';
export { default as FieldInput } from './FieldInput';
