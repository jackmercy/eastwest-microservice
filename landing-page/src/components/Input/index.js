export * from './components';

export { default } from './Input';
export { default as InputNumber } from './InputNumber';
export { default as InputPhone } from './InputPhone';
export { default as InputSearch } from './InputSearch';
export { default as InputPassword } from './InputPassword';
