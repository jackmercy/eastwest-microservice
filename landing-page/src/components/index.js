export { default as SEO } from './SEO';
export { default as Logo } from './Logo';
export { default as Modal } from './Modal';
export { default as Input } from './Input';
export { default as Select } from './Select';
export { default as Button } from './Button';
export { default as Loading } from './Loading';
export { default as Carousel } from './Carousel';
export { default as Textarea } from './Textarea';
export { default as CheckBox } from './CheckBox';
export { default as PopupMenu } from './PopupMenu';
export { default as StaticHTML } from './StaticHTML';

export * from './Input';
export * from './Modal';
