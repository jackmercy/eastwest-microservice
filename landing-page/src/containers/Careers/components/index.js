export { default as SectionTitle } from './SectionTitle';
export { default as MenuScrollableLayout } from './MenuScrollableLayout';
