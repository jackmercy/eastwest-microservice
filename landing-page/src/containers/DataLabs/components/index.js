export { default as Teaser } from './Teaser';
export { default as JoinUs } from './JoinUs';
export { default as Become } from './Become';
export { default as HowItWorks } from './HowItWorks';
export { default as ReadyToStart } from './ReadyToStart';
