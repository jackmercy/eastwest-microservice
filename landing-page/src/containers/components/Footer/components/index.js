export { default as Register } from './Register';
export { default as CopyRight } from './CopyRight';
export { default as SummaryInfo } from './SummaryInfo';
export { default as SectionTitle } from './SectionTitle';
export { default as SwitchLanguages } from './SwitchLanguages';
