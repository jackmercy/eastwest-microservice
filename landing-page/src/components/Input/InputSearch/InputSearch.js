import React, { memo } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import { useFormValidation } from '../../../hooks';
import { FieldLabel, FieldInput } from '../components';

import useStyles from './inputSearch.styles';

const InputSearch = ({
  label,
  onSearch,
  dataTest,
  required,
  className,
  inputClassName,
  labelClassName,
  ...restProps
}) => {
  const classes = useStyles();
  const {
    errors,
    formValues,
    handleResetFieldError,
    handleSetFieldValue,
    handleKeyDownEnter,
  } = useFormValidation({});

  const handleSearch = () => {
    onSearch(formValues.search);
  };

  return (
    <div
      className={className}
      data-test={dataTest || 'component-input-search-root'}
    >
      {label ? (
        <FieldLabel
          label={label}
          htmlFor="search"
          required={required}
          className={labelClassName}
        />
      ) : null}
      <div className={classes.inputWrapper}>
        <FieldInput
          {...restProps}
          type="text"
          name="search"
          hasError={!!errors.search}
          required={required}
          className={classNames(classes.input, inputClassName)}
          onChange={({ target: { name, value } }) =>
            handleSetFieldValue(name, value)}
          onFocus={() => handleResetFieldError('search')}
          onKeyDown={(e) => handleKeyDownEnter(e, handleSearch)}
        />
        <i
          onClick={handleSearch}
          role="presentation"
          className="icon-search"
          data-test="component-input-search-icon"
        />
      </div>
    </div>
  );
};

InputSearch.propTypes = {
  onSearch: PropTypes.func.isRequired,
  label: PropTypes.node,
  required: PropTypes.bool,
  dataTest: PropTypes.string,
  className: PropTypes.string,
  inputClassName: PropTypes.string,
  labelClassName: PropTypes.string,
};
InputSearch.defaultProps = {
  dataTest: '',
  className: '',
  required: false,
  label: undefined,
  inputClassName: '',
  labelClassName: '',
};

export default memo(InputSearch);
