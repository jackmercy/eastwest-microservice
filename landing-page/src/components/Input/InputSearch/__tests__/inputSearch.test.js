import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { FieldLabel, FieldInput } from '../../components';
import InputSearch from '../InputSearch';

describe('Testing Input component > InputSearch', () => {
  let wrapper;
  const props = {
    label: 'Search',
    name: 'search',
    error: 'some error',
    onSearch: jest.fn(),
  };

  beforeAll(() => {
    wrapper = mount(<InputSearch {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<InputSearch {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a FieldLabel', () => {
    expect(wrapper.find(FieldLabel)).toHaveLength(1);
  });

  it('should have a FieldInput', () => {
    expect(wrapper.find(FieldInput)).toHaveLength(1);
  });

  it('should have a search icon', () => {
    expect(
      wrapper.find('[data-test="component-input-search-icon"]'),
    ).toHaveLength(1);
  });
});
