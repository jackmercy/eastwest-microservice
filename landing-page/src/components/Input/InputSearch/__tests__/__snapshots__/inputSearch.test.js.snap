// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input component > InputSearch should render match snapshot 1`] = `
<div
  className=""
  data-test="component-input-search-root"
>
  <Memo(FieldLabel)
    className=""
    htmlFor="search"
    label="Search"
    required={false}
  />
  <div
    className="inputWrapper-0-2-1"
  >
    <Memo(FieldInput)
      className="input-0-2-2"
      error="some error"
      hasError={false}
      name="search"
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      required={false}
      type="text"
    />
    <i
      className="icon-search"
      data-test="component-input-search-icon"
      onClick={[Function]}
      role="presentation"
    />
  </div>
</div>
`;
